request_logging/middleware.py

```diff
@@ -133,20 +133,24 @@ def _log_multipart(self, body, logging_context):
         Splits multipart body into parts separated by "boundary", then matches each part to BINARY_REGEX
         which searches for existance of "Content-Type" and capture of what type is this part.
         If it is an image or an application replace that content with "(binary data)" string.
+        This function will log "(multipart/form)" if body can't be decoded by utf-8.
         """
-        body_str = body if isinstance(body, str) else body.decode()
-        parts = body_str.split(self.boundary)
-        last = len(parts) - 1
-        for i, part in enumerate(parts):
-            if 'Content-Type:' in part:
-                match = BINARY_REGEX.search(part)
-                if match and match.group(2) in BINARY_TYPES and not match.group(4) in ('', '\r\n'):
-                    part = match.expand(r'\1\2/\3\r\n\r\n(binary data)\r\n')
-
-            if i != last:
-                part = part + self.boundary
-
-            self.logger.log(self.log_level, part, logging_context)
+        try:
```

Comment:
Can we swap the try/ except for `isinstance(body, bytes)`? A quick comparison with `dis.dis` suggests that ("new" in the code below) would be faster than this ("current" below):

```
>>> dis(current)
  2           0 LOAD_CONST               1 (b'--BoUnDaRyStRiNg\r\n')
              3 STORE_FAST               0 (body)

  3           6 SETUP_EXCEPT            14 (to 23)

  4           9 LOAD_FAST                0 (body)
             12 LOAD_ATTR                0 (decode)
             15 CALL_FUNCTION            0 (0 positional, 0 keyword pair)
             18 POP_TOP
             19 POP_BLOCK
             20 JUMP_FORWARD            18 (to 41)

  5     >>   23 DUP_TOP
             24 LOAD_GLOBAL              1 (UnicodeDecodeError)
             27 COMPARE_OP              10 (exception match)
             30 POP_JUMP_IF_FALSE       40
             33 POP_TOP
             34 POP_TOP
             35 POP_TOP

  6          36 POP_EXCEPT
             37 JUMP_FORWARD             1 (to 41)
        >>   40 END_FINALLY
        >>   41 LOAD_CONST               0 (None)
             44 RETURN_VALUE
>>> dis(new)
  2           0 LOAD_CONST               1 (b'--BoUnDaRyStRiNg\r\n')
              3 STORE_FAST               0 (body)

  3           6 LOAD_GLOBAL              0 (isinstance)
              9 LOAD_FAST                0 (body)
             12 LOAD_GLOBAL              1 (bytes)
             15 CALL_FUNCTION            2 (2 positional, 0 keyword pair)
             18 POP_JUMP_IF_FALSE       25

  4          21 LOAD_CONST               0 (None)
             24 RETURN_VALUE
        >>   25 LOAD_CONST               0 (None)
             28 RETURN_VALUE
```